### ISSUE TITLE
Tests fail with OpenJDK 12

### ISSUE THREAD

**Issue by michelole** [jhalterman/typetools#52]
When trying to build the project with OpenJDK 12, tests fail with:

```
$ echo $JAVA_HOME 
/Library/Java/JavaVirtualMachines/openjdk-12.jdk/Contents/Home

$ java -version
openjdk version "12" 2019-03-19
OpenJDK Runtime Environment (build 12+33)
OpenJDK 64-Bit Server VM (build 12+33, mixed mode, sharing)

$ mvn test                                                                       
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------------< net.jodah:typetools >-------------------------
[INFO] Building TypeTools 0.6.2-SNAPSHOT

[...]

[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running TestSuite
[ERROR] Tests run: 100, Failures: 32, Errors: 0, Skipped: 0, Time elapsed: 1.403 s <<< FAILURE! - in TestSuite
[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0.036 s  <<< FAILURE!
java.lang.AssertionError: expected [class java.util.UUID] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.handlePassedFunction(LambdaTest.java:305)
	at net.jodah.typetools.functional.LambdaTest.shouldHandlePassedLambda(LambdaTest.java:289)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0 s  <<< FAILURE!
java.lang.AssertionError: expected [class java.util.UUID] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.handlePassedFunction(LambdaTest.java:305)
	at net.jodah.typetools.functional.LambdaTest.shouldHandlePassedLambda(LambdaTest.java:289)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0.003 s  <<< FAILURE!
java.lang.AssertionError: expected [class java.util.UUID] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.handlePassedFunction(LambdaTest.java:305)
	at net.jodah.typetools.functional.LambdaTest.shouldHandlePassedMethodRef(LambdaTest.java:300)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0.004 s  <<< FAILURE!
java.lang.AssertionError: expected [class java.util.UUID] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.handlePassedFunction(LambdaTest.java:305)
	at net.jodah.typetools.functional.LambdaTest.shouldHandlePassedMethodRef(LambdaTest.java:300)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0.002 s  <<< FAILURE!
java.lang.AssertionError: expected [class java.util.UUID] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.handlePassedFunction(LambdaTest.java:305)
	at net.jodah.typetools.functional.LambdaTest.shouldHandlePassedSerializableLambda(LambdaTest.java:293)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0 s  <<< FAILURE!
java.lang.AssertionError: expected [class java.util.UUID] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.handlePassedFunction(LambdaTest.java:305)
	at net.jodah.typetools.functional.LambdaTest.shouldHandlePassedSerializableLambda(LambdaTest.java:293)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0.003 s  <<< FAILURE!
java.lang.AssertionError: expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.shouldResolveArguments(LambdaTest.java:114)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0.002 s  <<< FAILURE!
java.lang.AssertionError: expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.shouldResolveArguments(LambdaTest.java:114)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0.003 s  <<< FAILURE!
java.lang.AssertionError: expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.shouldResolveArgumentsFromArbitraryObjectMethodRefs(LambdaTest.java:162)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0.001 s  <<< FAILURE!
java.lang.AssertionError: expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.shouldResolveArgumentsFromArbitraryObjectMethodRefs(LambdaTest.java:162)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0.001 s  <<< FAILURE!
java.lang.AssertionError: expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.shouldResolveArgumentsFromInstanceMethodRefs(LambdaTest.java:142)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0.001 s  <<< FAILURE!
java.lang.AssertionError: expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.shouldResolveArgumentsFromInstanceMethodRefs(LambdaTest.java:142)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0 s  <<< FAILURE!
java.lang.AssertionError: Lists differ at element [0]: class java.lang.String != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.shouldResolveArgumentsFromNonSamMethodRef(LambdaTest.java:174)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0.001 s  <<< FAILURE!
java.lang.AssertionError: Lists differ at element [0]: class java.lang.String != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.shouldResolveArgumentsFromNonSamMethodRef(LambdaTest.java:174)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0 s  <<< FAILURE!
java.lang.AssertionError: expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.shouldResolveArgumentsFromStaticMethodRefs(LambdaTest.java:151)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0.001 s  <<< FAILURE!
java.lang.AssertionError: expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.shouldResolveArgumentsFromStaticMethodRefs(LambdaTest.java:151)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0.001 s  <<< FAILURE!
java.lang.AssertionError: Lists differ at element [0]: class net.jodah.typetools.TypeResolver$Unknown != class java.lang.String expected [class net.jodah.typetools.TypeResolver$Unknown] but found [class java.lang.String]
	at net.jodah.typetools.functional.LambdaTest.shouldResolveCapturedArguments(LambdaTest.java:131)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0.001 s  <<< FAILURE!
java.lang.AssertionError: Lists differ at element [0]: class net.jodah.typetools.TypeResolver$Unknown != class java.lang.String expected [class net.jodah.typetools.TypeResolver$Unknown] but found [class java.lang.String]
	at net.jodah.typetools.functional.LambdaTest.shouldResolveCapturedArguments(LambdaTest.java:131)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0.003 s  <<< FAILURE!
java.lang.AssertionError: Lists differ at element [0]: class java.lang.String != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.shouldResolveMultiArguments(LambdaTest.java:206)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0.001 s  <<< FAILURE!
java.lang.AssertionError: Lists differ at element [0]: class java.lang.String != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.shouldResolveMultiArguments(LambdaTest.java:206)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0.003 s  <<< FAILURE!
java.lang.AssertionError: Lists differ at element [0]: class java.lang.String != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.shouldResolveMultiArgumentsForMethodRefs(LambdaTest.java:228)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0.001 s  <<< FAILURE!
java.lang.AssertionError: Lists differ at element [0]: class java.lang.String != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.shouldResolveMultiArgumentsForMethodRefs(LambdaTest.java:228)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0.002 s  <<< FAILURE!
java.lang.AssertionError: Lists differ at element [0]: class java.lang.String != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.shouldResolveSubclassArguments(LambdaTest.java:241)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0.001 s  <<< FAILURE!
java.lang.AssertionError: Lists differ at element [0]: class java.lang.String != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.shouldResolveSubclassArguments(LambdaTest.java:241)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0.001 s  <<< FAILURE!
java.lang.AssertionError: Lists differ at element [0]: class java.lang.String != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.shouldResolveSubclassArgumentsForConstructor(LambdaTest.java:247)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0 s  <<< FAILURE!
java.lang.AssertionError: Lists differ at element [0]: class java.lang.String != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.shouldResolveSubclassArgumentsForConstructor(LambdaTest.java:247)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0.001 s  <<< FAILURE!
java.lang.AssertionError: Lists differ at element [0]: class java.lang.String != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.shouldResolveSubclassArgumentsForConstructorRef(LambdaTest.java:262)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0.001 s  <<< FAILURE!
java.lang.AssertionError: Lists differ at element [0]: class java.lang.String != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.shouldResolveSubclassArgumentsForConstructorRef(LambdaTest.java:262)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0.002 s  <<< FAILURE!
java.lang.AssertionError: Lists differ at element [0]: class java.lang.String != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.shouldResolveSubclassArgumentsForMethodRefs(LambdaTest.java:256)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0 s  <<< FAILURE!
java.lang.AssertionError: Lists differ at element [0]: class java.lang.String != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.shouldResolveSubclassArgumentsForMethodRefs(LambdaTest.java:256)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0.007 s  <<< FAILURE!
java.lang.AssertionError: Lists differ at element [0]: class java.lang.Integer != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.Integer] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.shouldResolveTransposedSubclassArguments(LambdaTest.java:268)

[ERROR] (net.jodah.typetools.functional.LambdaTest)  Time elapsed: 0 s  <<< FAILURE!
java.lang.AssertionError: Lists differ at element [0]: class java.lang.Integer != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.Integer] but found [class net.jodah.typetools.TypeResolver$Unknown]
	at net.jodah.typetools.functional.LambdaTest.shouldResolveTransposedSubclassArguments(LambdaTest.java:268)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR] net.jodah.typetools.functional.LambdaTest.(net.jodah.typetools.functional.LambdaTest)
[ERROR]   Run 1: LambdaTest.shouldHandlePassedLambda:289->handlePassedFunction:305 expected [class java.util.UUID] but found [class net.jodah.typetools.TypeResolver$Unknown]
[ERROR]   Run 2: LambdaTest.shouldHandlePassedLambda:289->handlePassedFunction:305 expected [class java.util.UUID] but found [class net.jodah.typetools.TypeResolver$Unknown]
[ERROR]   Run 3: LambdaTest.shouldHandlePassedMethodRef:300->handlePassedFunction:305 expected [class java.util.UUID] but found [class net.jodah.typetools.TypeResolver$Unknown]
[ERROR]   Run 4: LambdaTest.shouldHandlePassedMethodRef:300->handlePassedFunction:305 expected [class java.util.UUID] but found [class net.jodah.typetools.TypeResolver$Unknown]
[ERROR]   Run 5: LambdaTest.shouldHandlePassedSerializableLambda:293->handlePassedFunction:305 expected [class java.util.UUID] but found [class net.jodah.typetools.TypeResolver$Unknown]
[ERROR]   Run 6: LambdaTest.shouldHandlePassedSerializableLambda:293->handlePassedFunction:305 expected [class java.util.UUID] but found [class net.jodah.typetools.TypeResolver$Unknown]
[ERROR]   Run 7: LambdaTest.shouldResolveArguments:114 expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
[ERROR]   Run 8: LambdaTest.shouldResolveArguments:114 expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
[ERROR]   Run 9: LambdaTest.shouldResolveArgumentsFromArbitraryObjectMethodRefs:162 expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
[ERROR]   Run 10: LambdaTest.shouldResolveArgumentsFromArbitraryObjectMethodRefs:162 expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
[ERROR]   Run 11: LambdaTest.shouldResolveArgumentsFromInstanceMethodRefs:142 expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
[ERROR]   Run 12: LambdaTest.shouldResolveArgumentsFromInstanceMethodRefs:142 expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
[ERROR]   Run 13: LambdaTest.shouldResolveArgumentsFromNonSamMethodRef:174 Lists differ at element [0]: class java.lang.String != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
[ERROR]   Run 14: LambdaTest.shouldResolveArgumentsFromNonSamMethodRef:174 Lists differ at element [0]: class java.lang.String != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
[ERROR]   Run 15: LambdaTest.shouldResolveArgumentsFromStaticMethodRefs:151 expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
[ERROR]   Run 16: LambdaTest.shouldResolveArgumentsFromStaticMethodRefs:151 expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
[ERROR]   Run 17: LambdaTest.shouldResolveCapturedArguments:131 Lists differ at element [0]: class net.jodah.typetools.TypeResolver$Unknown != class java.lang.String expected [class net.jodah.typetools.TypeResolver$Unknown] but found [class java.lang.String]
[ERROR]   Run 18: LambdaTest.shouldResolveCapturedArguments:131 Lists differ at element [0]: class net.jodah.typetools.TypeResolver$Unknown != class java.lang.String expected [class net.jodah.typetools.TypeResolver$Unknown] but found [class java.lang.String]
[ERROR]   Run 19: LambdaTest.shouldResolveMultiArguments:206 Lists differ at element [0]: class java.lang.String != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
[ERROR]   Run 20: LambdaTest.shouldResolveMultiArguments:206 Lists differ at element [0]: class java.lang.String != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
[ERROR]   Run 21: LambdaTest.shouldResolveMultiArgumentsForMethodRefs:228 Lists differ at element [0]: class java.lang.String != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
[ERROR]   Run 22: LambdaTest.shouldResolveMultiArgumentsForMethodRefs:228 Lists differ at element [0]: class java.lang.String != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
[ERROR]   Run 23: LambdaTest.shouldResolveSubclassArguments:241 Lists differ at element [0]: class java.lang.String != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
[ERROR]   Run 24: LambdaTest.shouldResolveSubclassArguments:241 Lists differ at element [0]: class java.lang.String != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
[ERROR]   Run 25: LambdaTest.shouldResolveSubclassArgumentsForConstructor:247 Lists differ at element [0]: class java.lang.String != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
[ERROR]   Run 26: LambdaTest.shouldResolveSubclassArgumentsForConstructor:247 Lists differ at element [0]: class java.lang.String != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
[ERROR]   Run 27: LambdaTest.shouldResolveSubclassArgumentsForConstructorRef:262 Lists differ at element [0]: class java.lang.String != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
[ERROR]   Run 28: LambdaTest.shouldResolveSubclassArgumentsForConstructorRef:262 Lists differ at element [0]: class java.lang.String != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
[ERROR]   Run 29: LambdaTest.shouldResolveSubclassArgumentsForMethodRefs:256 Lists differ at element [0]: class java.lang.String != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
[ERROR]   Run 30: LambdaTest.shouldResolveSubclassArgumentsForMethodRefs:256 Lists differ at element [0]: class java.lang.String != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.String] but found [class net.jodah.typetools.TypeResolver$Unknown]
[ERROR]   Run 31: LambdaTest.shouldResolveTransposedSubclassArguments:268 Lists differ at element [0]: class java.lang.Integer != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.Integer] but found [class net.jodah.typetools.TypeResolver$Unknown]
[ERROR]   Run 32: LambdaTest.shouldResolveTransposedSubclassArguments:268 Lists differ at element [0]: class java.lang.Integer != class net.jodah.typetools.TypeResolver$Unknown expected [class java.lang.Integer] but found [class net.jodah.typetools.TypeResolver$Unknown]
[INFO] 
[INFO] 
[ERROR] Tests run: 69, Failures: 1, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  11.381 s
[INFO] Finished at: 2019-03-26T18:48:56+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M3:test (default-test) on project typetools: There are test failures.
[...]
```

**Comment by szpak** [comment]
mockito-java8 which depends on typetools for some kind of operations also fails with Java 12: https://github.com/szpak/mockito-java8/issues/33

**Comment by jhalterman** [comment]
Here we go again :)

Happy to take a PR for this.

**Comment by jhalterman** [comment]
Closed by #66.
